front_end/core/host/RNPerfMetrics.ts

```diff
@@ -141,6 +168,11 @@ function getPerfTimestamp(): DOMHighResTimeStamp {
   return performance.timeOrigin + performance.now();
 }
 
+function maybeTruncateDeveloperResourceUrl(parsedURL: ParsedURL): string {
+  const {url} = parsedURL;
+  return parsedURL.isHttpOrHttps() ? url : `${url.slice(0, 100)} …(omitted ${url.length - 100} characters)`;
```

Comment:
Is there an example/reason for why we'd want to truncate? :)

Reply:
Yes. Some of these are `data:` urls which can go as high as 7MB; here's a real-life example:

```
data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi9fX3ByZWx1ZGVfXyIsIi9bb …(omitted 6979931 characters)
```

We mostly care about http urls only, so those are not truncated, as that's the place we need more visibility, i.e. the round trip from CDT to Metro/device.

Can add this context as an inline comment if preferred?